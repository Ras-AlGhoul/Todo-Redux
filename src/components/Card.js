import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  markDone,
  edit,
  setEditInput,
} from '../reducers/actions';

const Card = ({ task }) => {
  const dispatch = useDispatch();
  const editInputReducer = useSelector(state => state.theEditInputReducer);
  const [toggle, setToggle] = useState(false);

  const handleClickDone = () => {
    dispatch(markDone(task));
  };
  const handleEdit = e => {
    e.preventDefault();
    dispatch(edit(editInputReducer));
    setToggle(false);
    dispatch(setEditInput({ todo: '', previousTask: '' }));
  };

  return (
    <ul data-testid='card' className='card'>
      {!toggle && (
        <li className="done-card__title"> {task} </li>
      )}
      {toggle && (
        <form onSubmit={handleEdit}>
          <input className='card__input' value={editInputReducer.todo} onChange={({ target: { value } }) => dispatch(setEditInput({ todo: value, previousTask: task }))} />
          <button className='card__btn' type='submit'>Cofirm</button>
        </form>
      )}
      <button className="card__btn1" onClick={handleClickDone}>Done</button> <button className="card__btn2" onClick={() => setToggle(true)}> Edit </button>
    </ul>
  );
};

export default Card;
