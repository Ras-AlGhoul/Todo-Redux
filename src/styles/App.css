* {
  box-sizing: border-box;
}

body {
  font-family: Arial, Helvetica, sans-serif;
  margin: 0 auto;
  background-image: url("../assets/3.jpg");
  padding: 40px;
  max-width: 900 px;
  background-size: cover;
  text-align: center;
}

ul{
  list-style: none;
}

.form{
  text-align: left;
  padding: 35px 20px;
  background-color:rgba(154, 143, 137, 0.626) ;
  border-radius: 25px;
  height: 180px;
}

.form__label{
  font-size: 40px;
}

.form__input{
  height: 20px;
  height: 40px;
  width: 100%;
}
.form__btn {
  float: right;
  margin-top: 16px;
  border: 1px solid black;
  background: none;
  height: 30px;
  width: 60px;
  border-radius: 7px;
  color: black;
}

.form__btn:hover{
  color: azure;
  background-color: black;
}

.card{
  background-color: #b8b8b889;
  text-align: left;
  padding: 35px 20px;
  border-radius: 25px;
  margin-top: 40px;
}

.card__title{
  font-size: 20px;
}

.done-card{
  text-align: left;
  text-decoration: line-through;
  padding: 35px 20px;
  border-radius: 25px;
  margin-top: 40px;
  background-color: #39A388;
  bottom: 0px;
  list-style: none;
}

.card__removbtn{
  float: right;
  height: 30px;
  width: 60px;
  border: 1px solid red;
  background: none;
  border-radius: 7px 0px 0px 7px;
  color: red;
}

.done-card__title{
  font-size: 30px;
}

.card__removbtn:hover{
  color: azure;
  background-color: red;
}

.card__btn1{
  float: right;
  border: 1px solid black;
  background: none;
  height: 30px;
  width: 60px;
  border-radius: 0px 7px 7px 0px;
  color: black;
}
.card__btn1:hover{
  color: azure;
  background-color: black;
}
.card__btn2{
  float: right;
  border: 1px solid black;
  background: none;
  height: 30px;
  width: 60px;
  border-radius: 7px 0px 0px 7px;
  color: black;
}
.card__btn2:hover{
  color: azure;
  background-color: black;
}
.app__btn{
  margin-top: 16px;
  border: 1px solid black;
  background: black;
  height: 30px;
  width: 160px;
  border-radius: 7px;
  color: white;
}
.app__btn:hover{
  color: black;
  background-color:transparent;
}

.card__btn{
  margin-top: 16px;
  border: 1px solid black;
  background: none;
  height: 30px;
  width: 70px;
  border-radius: 7px;
  color: black;
}
.card__btn:hover{
  color: azure;
  background-color: black;
}

.card__input{
  font-size: 30px;
  height: 20px;
  height: 40px;
  width: 100%;
}