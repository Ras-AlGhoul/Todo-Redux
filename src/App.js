import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import './styles/App.css';
import Form from './components/Form';
import Card from './components/Card';
import DoneCard from './components/DoneCard';
import { setToggle } from './reducers/actions';

const App = () => {
  const dispatch = useDispatch();
  const listReducer = useSelector(state => state.listReducer);
  const toggleReducer = useSelector(state => state.toggleReducer);
  localStorage.setItem('todoList', JSON.stringify(listReducer.todos));
  localStorage.setItem('doneTodoList', JSON.stringify(listReducer.doneTodos));
  const todosList = JSON.parse(localStorage.getItem('todoList'));
  const doneTodoList = JSON.parse(localStorage.getItem('doneTodoList'));

  return (
    <div className='App' data-testid='app'>
      <Form />
      {todosList.map((item, index) => <Card
        key={index}
        task={item}
      />).reverse()
      }
      <button className='app__btn' onClick={() => dispatch(setToggle(true))}>Unhide/Hide done tasks</button>
      {toggleReducer && (
        doneTodoList.map((item, index) => <DoneCard
          key={index}
          task={item}
        />).reverse()
      )
      }
    </div>
  );
};

export default App;
