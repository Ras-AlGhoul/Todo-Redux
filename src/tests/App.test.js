import { render, screen, fireEvent, waitFor, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import App from '../App';
import Card from '../components/Card';
const state = {  
  todos: ['buy Milk', 'buy food', 'finish course', 'go to the gym'],
};

afterEach(() => {
  cleanup()
});

describe('App Element Testing', () => {
  test('should be in the document', () => {
    render(<App />);
    const AppElement = screen.getByTestId('app');
    expect(AppElement).toBeInTheDocument();
  });

  test('should contain form components', () => {
    render(<App todos={state.todos}/>);
    const AppElement = screen.getByTestId('app');
    const formElement = screen.getByTestId('form');
    expect(AppElement).toContainElement(formElement);
  });

  test('should not have li Element', () => {
    render(<App />);
    const AppElement = screen.getByTestId('app');
    expect(AppElement).not.toContainHTML('<li></li>');
  });

  test('should match snapshot', () => {
      const snapshot = renderer.create(<App />).toJSON();
      expect(snapshot).toMatchSnapshot();
  });
});