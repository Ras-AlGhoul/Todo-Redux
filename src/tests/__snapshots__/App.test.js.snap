// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Element Testing should match snapshot 1`] = `
<div
  className="App"
  data-testid="app"
>
  <form
    className="form"
    data-testid="form"
    onSubmit={[Function]}
  >
    <label>
       Title 
    </label>
    <input
      onChange={[Function]}
      placeholder="Add a task..."
      type="text"
      value=""
    />
    <button
      className="form__btn"
      type="submit"
    >
       Add 
    </button>
  </form>
</div>
`;
